src/crowdin_gui.cpp

```diff
@@ -676,6 +677,28 @@ bool ExtractCrowdinMetadata(CatalogPtr cat,
                 : cat->GetLanguage();
     }
 
+    if (const auto xliff = dynamic_cast<XLIFFCatalog*>(cat.get()))
+    {
+        auto xml = xliff->GetXMLRoot();
+        int projId = xml.attribute("project-id").as_int(-1);
+        if (projId > 0)
+        {
+            auto files = xml.children("file");
+            if (files.begin() != files.end())
+            {
+                int _fileId = files.begin()->attribute("id").as_int(-1);
+                if (_fileId > 0)
+                {
+                    if (projectId)
+                        *projectId = projId;
+                    if (fileId)
+                        *fileId = _fileId;
+                    return true;
+                }
+            }
```

Comment:
This part, being relatively expensive, should be gated by `if (fileId)`

Reply:
Optimized https://github.com/vslavik/poedit/pull/644/commits/45f90046cdbc69a8773fb2c6c69ee6c3a3e9f5f2#diff-a1865a3025926eb82afbd7f4d9f13518R684 that attribures are not converted to integers in case they should not be stored to `projId` and `fileId` but IMO they _both_ need to be checked for presence in any case since without it we can not surely treat that XLIFF file as "from Crowdin" and `return true` (i.e. if at least either project or file ID is present). Unless I miss something or if you have better ideas how to optimize it but keeping same level of insurance.